### PR TITLE
Add ListEntryCommand

### LINKED ISSUE #70: Refactor code so that schedule and tasks are under one class, Task

### PR DESCRIPTION
**Pull request by lrj689**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListEntryCommand.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Lists all entries in address book based on the filter provided
+ */
+public class ListEntryCommand extends Command {
+    public static final String COMMAND_WORD = "elist";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Lists all entries "
+            + "by displaying them as a list sorted by date. Entries can also be listed by day/week.\n"
+            + "Possible parameters: day/week \n"
```
Comment (tsh22) [inline src/main/java/seedu/address/logic/commands/ListEntryCommand.java]: There's the option to just use elist without the parameters also right? Perhaps can update the message here

**Review by tsh22 — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/ListEntryCommand.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Lists all entries in address book based on the filter provided
+ */
+public class ListEntryCommand extends Command {
+    public static final String COMMAND_WORD = "elist";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Lists all entries "
+            + "by displaying them as a list sorted by date. Entries can also be listed by day/week.\n"
+            + "Optional parameters: day/week \n"
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/logic/commands/ListEntryCommand.java]: let's use [day/week] so it's uniform with user guide

File: src/main/java/seedu/address/logic/parser/ListEntryCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * Parses the give {@code String} of arguments in the context of the ListEntryCommand
+     * and returns a ListEntryCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListEntryCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        ListEntryCommand listEntryCommand = new ListEntryCommand(
+                new ListEntryFormatPredicate(trimmedArgs));
+        if (trimmedArgs.equals("day") || trimmedArgs.equals("week") || trimmedArgs.isEmpty()) {
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/logic/parser/ListEntryCommandParser.java:21]: we should check trimmedArgs first before we instantiate a ListEntryCommand. would going with something like this be better?
```java
if (trimmedArgs.equals("day") || trimmedArgs.equals("week") || trimmedArgs.isEmpty()) {
    return new ListEntryCommand(new ListEntryFormatPredicate(trimmedArgs));
} else {
    throw new ParseException(
            String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListEntryCommand.MESSAGE_USAGE));
}
```

**Review by nicholastanvis — CHANGES_REQUESTED** [review]

**Review by goatygoatygoat — APPROVED** [review]

**Review by nicholastanvis — APPROVED** [review]
comment addressed, lgtm

**Pull request merged**